linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -1808,15 +1886,41 @@ private[emitter] object CoreJSLib {
                 genArrowFunction(paramList(that), Return(self === that))
               }),
               privateFieldSet(cpn.wrapArray, {
-                If(typedArrayClass, {
-                  genArrowFunction(paramList(array), {
-                    Return(New(arrayClass, New(typedArrayClass, array :: Nil) :: Nil))
+                val whenNotRuntimeLongArray = {
+                  If(typedArrayClass, {
+                    genArrowFunction(paramList(array), {
+                      Return(New(arrayClass, New(typedArrayClass, array :: Nil) :: Nil))
+                    })
+                  }, {
+                    genArrowFunction(paramList(array), {
+                      Return(New(arrayClass, array :: Nil))
+                    })
                   })
-                }, {
-                  genArrowFunction(paramList(array), {
-                    Return(New(arrayClass, array :: Nil))
+                }
+                if (useBigIntForLongs) {
+                  whenNotRuntimeLongArray
+                } else {
+                  If(isLongArray, {
+                    genArrowFunction(paramList(array), {
+                      val len = varRef("len")
+                      val result = varRef("result")
+                      val u = varRef("u")
+                      val i = varRef("i")
+                      Block(
+                        const(len, array.length | 0),
+                        const(result, New(arrayClass, len :: Nil)),
+                        const(u, result.u),
+                        For(let(i, 0), i < len, i := (i + 1) | 0, Block(
+                          BracketSelect(u, i << 1) := BracketSelect(array, i) DOT cpn.lo,
+                          BracketSelect(u, ((i << 1) + 1) | 0) := BracketSelect(array, i) DOT cpn.hi
```

Comment:
```suggestion
                          const(srcElem, BracketSelect(array, i))
                          BracketSelect(u, i << 1) := srcElem DOT cpn.lo,
                          BracketSelect(u, ((i << 1) + 1) | 0) := srcElem DOT cpn.hi
```

For speed? (not sure it's measurable, but it seems we're usually applying this pattern).